From 238379428210e0bac9f078c0c27ff320931a5024 Mon Sep 17 00:00:00 2001
From: Jacob Young <jacobly0@users.noreply.github.com>
Date: Tue, 4 Nov 2025 15:47:27 -0500
Subject: [PATCH 043/133] x86_64: implement split vector stores
X-Developer-Signature: v=1; a=openpgp-sha256; l=1108; i=xtex@astrafall.org;
 h=from:subject; bh=6zaxG6AOp9nHDYVOT5kZuHt2AQ/m8Aw40/tiRWLYHUU=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn7cv9v1K47k6dtwntCIOBQ0FEz0YdCh3a+lnSarPZjP
 YPu7H0ZHRNZGMS4GCzFFFmKDBu8WXXS+UWXlcvCzGFlAhkiLdLAwMDAwMLAl5uYV2qkY6Rnqm2o
 Z2ioY6RjxMDFKQBTLf+Z4X/xba5EceXA1ILzwb/+2asbuE45UrDx7OI3kVaWTiFdN6czMsz5Ne/
 3vzXtoZn3z01nzryUuNuo2VZ/j9kv722qLlZebcwA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Closes #25809

(cherry picked from commit a6d444c2714f24b7232895cf15282e2287fe445e)
---
 src/arch/x86_64/CodeGen.zig | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/arch/x86_64/CodeGen.zig b/src/arch/x86_64/CodeGen.zig
index 39997de7fbf6..54ca27eb3096 100644
--- a/src/arch/x86_64/CodeGen.zig
+++ b/src/arch/x86_64/CodeGen.zig
@@ -186989,6 +186989,10 @@ const Temp = struct {
                     assert(src_regs.len - part_index == std.math.divCeil(u32, src_abi_size, 8) catch unreachable);
                     break :part_ty try cg.pt.intType(.unsigned, @as(u16, 8) * @min(src_abi_size, 8));
                 },
+                .vector_type => |vector_type| try cg.pt.vectorType(.{
+                    .len = @intCast(@divExact(vector_type.len, src_regs.len)),
+                    .child = vector_type.child,
+                }),
                 .opt_type => |opt_child| switch (ip.indexToKey(opt_child)) {
                     else => std.debug.panic("{s}: {f}\n", .{ @src().fn_name, src_ty.fmt(cg.pt) }),
                     .ptr_type => |ptr_info| {
-- 
2.52.0

